Add `monitoring` blocks to Identity Platform Config and Tenant resources

bschaatsbergen opened the pull request:
Closes [#13326](https://github.com/hashicorp/terraform-provider-google/issues/13326)

<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->




<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none
    
Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
identity_platform: Added attribute block `monitoring` to `google_identity_platform_tenant`
```

```release-note:enhancement
identity_platform: Added attribute block `monitoring` to `google_identity_platform_config`
```

```release-note:enhancement
identity_platform: Added attribute block `multi_tenant_config` to `google_identity_platform_config`
```

modular-magician commented:
Oops! It looks like you're using an unknown release-note type in your changelog entries:
* REPLACEME

Please only use the types listed in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md.

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I've detected that you're a community contributor. @slevenick, a repository maintainer, has been assigned to assist you and help review your changes. 

<details>
  <summary>:question: First time contributing? Click here for more details</summary>

---

Your assigned reviewer will help review your code by: 
* Ensuring it's backwards compatible, covers common error cases, etc.
* Summarizing the change into a user-facing changelog note.
* Passes tests, either our "VCR" suite, a set of presubmit tests, or with manual test runs.

You can help make sure that review is quick by running local tests and ensuring they're passing in between each push you make to your PR's branch. Also, try to leave a comment with each push you make, as pushes generally don't generate emails.

If your reviewer doesn't get back to you within a week after your most recent change, please feel free to leave a comment on the issue asking them to take a look! In the absence of a dedicated review dashboard most maintainers manage their pending reviews through email, and those will sometimes get lost in their inbox.

---

</details>

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 411 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 411 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 99 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 111 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2375` 
Passed tests `2123` 
Skipped tests: `249` 
Failed tests: `3` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccLoggingBucketConfigProject_cmekSettings|TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample|TestAccContainerCluster_withInvalidGatewayApiConfigChannel </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccLoggingBucketConfigProject_cmekSettings`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/1ac8b06f-f68b-4138-b5c0-a255cbfb5ad4/recording/TestAccLoggingBucketConfigProject_cmekSettings.log)]
`TestAccContainerCluster_withInvalidGatewayApiConfigChannel`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/1ac8b06f-f68b-4138-b5c0-a255cbfb5ad4/recording/TestAccContainerCluster_withInvalidGatewayApiConfigChannel.log)] 

Tests failed during RECORDING mode:
 `TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/1ac8b06f-f68b-4138-b5c0-a255cbfb5ad4/build-log/recording_build/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/1ac8b06f-f68b-4138-b5c0-a255cbfb5ad4/recording/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/1ac8b06f-f68b-4138-b5c0-a255cbfb5ad4/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/1ac8b06f-f68b-4138-b5c0-a255cbfb5ad4/recording) for each test

bschaatsbergen commented:
Hi @slevenick, I believe that most is done for this PR. Could you please share the error logs of the failed acceptance test?

bschaatsbergen commented:
Hi @slevenick, could you please share the error log from the failed test. I can't seem to run the test locally for some reason.

bschaatsbergen commented:
/gcbrun

slevenick commented on mmv1/templates/terraform/examples/identity_platform_tenant_monitoring.tf.erb:
```diff
@@ -0,0 +1,10 @@
+resource "google_identity_platform_tenant" "<%= ctx[:primary_resource_id] %>" {
```
Looks like the error is with this resource, the project is being defaulted to the project that all the CI tests run in rather than a new one which causes it to fail with:

=== CONT  TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample
    provider_test.go:315: Step 1/2 error: Error running apply: exit status 1
        
        Error: Error creating Tenant: googleapi: Error 400: INVALID_PROJECT_ID
        
          with google_identity_platform_tenant.tenant,
          on terraform_plugin_test.tf line 3, in resource "google_identity_platform_tenant" "tenant":
           3: resource "google_identity_platform_tenant" "tenant" {

bschaatsbergen replied:
Hmm right, what do you suggest I do in this case?

slevenick replied:
That error happens because in the response the monitoring block isn't included for some reason. Do you know why this would be?
```
{
  "name": "projects/my-project/tenants/tenant-25ocl",
  "displayName": "tenant",
  "allowPasswordSignup": true,
  "inheritance": {}
}
```

bschaatsbergen replied:
That's very weird, the response should contain the monitoring block, https://cloud.google.com/identity-platform/docs/reference/rest/v2/projects.tenants/patch#response-body

bschaatsbergen commented on mmv1/products/identityplatform/terraform.yaml:
```diff
@@ -28,6 +28,16 @@ overrides: !ruby/object:Overrides::ResourceOverrides
+        test_env_vars:
+          org_id: :ORG_ID
+          billing_acct: :BILLING_ACCT
+        # Resource creation race
```
I copied this line from the example above, they seem to be similar: https://github.com/GoogleCloudPlatform/magic-modules/blob/f6c18a8db48c01f8f327df4a269df6e9bbb0a599/mmv1/products/identityplatform/terraform.yaml#L21-L40

slevenick replied:
That's because that test has two instances of google_project_service. Even that should be fixable.

Can you remove skip_vcr and instead add the following to the google_identity_platform_config resource within that test:
depends_on = [google_project_service.identitytoolkit]

bschaatsbergen replied:
Done, https://github.com/GoogleCloudPlatform/magic-modules/pull/7025/commits/762a4defaad78cbabf2b7c235974d650604c0ef6

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 416 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 416 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 99 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 111 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2391` 
Passed tests `2136` 
Skipped tests: `248` 
Failed tests: `7` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccDatastreamStream_update|TestAccDatastreamStream_datastreamStreamFullExample|TestAccDatastreamStream_datastreamStreamBasicExample|TestAccContainerCluster_withInvalidGatewayApiConfigChannel|TestAccFirebaserulesRelease_BasicRelease|TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample|TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccDatastreamStream_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording/TestAccDatastreamStream_update.log)]
`TestAccDatastreamStream_datastreamStreamFullExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording/TestAccDatastreamStream_datastreamStreamFullExample.log)]
`TestAccDatastreamStream_datastreamStreamBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording/TestAccDatastreamStream_datastreamStreamBasicExample.log)]
`TestAccContainerCluster_withInvalidGatewayApiConfigChannel`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording/TestAccContainerCluster_withInvalidGatewayApiConfigChannel.log)]
`TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording/TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample.log)] 

Tests failed during RECORDING mode:
 `TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/build-log/recording_build/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/548fe296-7af1-401f-a8c2-8387826c17eb/recording) for each test

bschaatsbergen commented:
Looks like `TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample` now passes, just `TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample` that fails @slevenick

slevenick commented:
> Looks like `TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample` now passes, just `TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample` that fails @slevenick

Ok looks like the previous error was temporary, and now it's just the error due to the field not being returned:
```
          # google_identity_platform_tenant.tenant will be updated in-place
          ~ resource "google_identity_platform_tenant" "tenant" {
                id                       = "projects/project/tenants/tenant-0mnu0"
                name                     = "tenant-0mnu0"
                # (5 unchanged attributes hidden)
        
              + monitoring {
                  + request_logging {
                      + enabled = true
                    }
                }
            }
```

I worry that these fields may not be set correctly if they aren't returning from the API

bschaatsbergen commented:
@slevenick, I'm going to run the test locally tonight, were you meanwhile able to see anything that might cause the issue?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 416 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 416 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 99 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 111 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2421` 
Passed tests `2166` 
Skipped tests: `252` 
Failed tests: `3` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample|TestAccFirebaserulesRelease_BasicRelease|TestAccRegionInstanceGroupManager_stateful </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/70db610c-fffb-4d67-acbb-e278084bcc96/recording/TestAccFirebaserulesRelease_BasicRelease.log)] 

Tests failed during RECORDING mode:
 `TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/70db610c-fffb-4d67-acbb-e278084bcc96/build-log/recording_build/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/70db610c-fffb-4d67-acbb-e278084bcc96/recording/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample.log)]
`TestAccRegionInstanceGroupManager_stateful`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/70db610c-fffb-4d67-acbb-e278084bcc96/build-log/recording_build/TestAccRegionInstanceGroupManager_stateful_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/70db610c-fffb-4d67-acbb-e278084bcc96/recording/TestAccRegionInstanceGroupManager_stateful.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/70db610c-fffb-4d67-acbb-e278084bcc96/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/70db610c-fffb-4d67-acbb-e278084bcc96/recording) for each test

slevenick commented:
> @slevenick, I'm going to run the test locally tonight, were you meanwhile able to see anything that might cause the issue?

It continues to be that Terraform is not receiving this field from the API, and so detects a diff after apply. The solution is likely to add ignore_read on the field, as it seems that we can't read the value from the API

bschaatsbergen commented:
After some digging by a colleague of mine it seems that multi tenancy must be enabled for the `google_identity_platform_tentant` resource, which is made possible via 'multiTenantConfig' - an attribute block that wasn't present on the resource yet.

![image](https://user-images.githubusercontent.com/58337159/213014202-2aef24db-5ad4-42bf-bcfe-79882f6fbaae.png)

I've added the attribute block, and updated the test. Can you see why the generate diff fails, I can generate the provider locally without a problem. @slevenick 

thanks for your finding @laurensknoll.

slevenick commented on mmv1/products/identityplatform/terraform.yaml:
```diff
@@ -109,6 +117,9 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         name: "identity_platform_tenant_basic"
         primary_resource_id: "tenant"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "identity_platform_tenant_monitoring"
```
The file below references fields that aren't specified in this example block, causing the error.

The .tf.erb file references org_id & billing_acct

bschaatsbergen replied:
Check, thanks - fixed.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 588 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 588 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 139 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2422` 
Passed tests `2168` 
Skipped tests: `252` 
Failed tests: `2` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample|TestAccFirebaserulesRelease_BasicRelease </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/8e9058b8-4d8b-4d80-bee7-288a27056fe4/recording/TestAccFirebaserulesRelease_BasicRelease.log)] 

Tests failed during RECORDING mode:
 `TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/8e9058b8-4d8b-4d80-bee7-288a27056fe4/build-log/recording_build/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/8e9058b8-4d8b-4d80-bee7-288a27056fe4/recording/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/8e9058b8-4d8b-4d80-bee7-288a27056fe4/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/8e9058b8-4d8b-4d80-bee7-288a27056fe4/recording) for each test

bschaatsbergen commented:
Same issue @slevenick ?

slevenick commented:
Now it's new!
```
        [
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "Invalid JSON payload received. Unknown name \"multiTenantConfig\" at 'config': Cannot find field.",
                "field": "config"
              }
            ]
          }
        ]
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 588 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7025-old..auto-pr-7025) ( 6 files changed, 588 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7025-old..auto-pr-7025) ( 4 files changed, 139 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2425` 
Passed tests `2170` 
Skipped tests: `252` 
Failed tests: `3` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample|TestAccFirebaserulesRelease_BasicRelease|TestAccRegionInstanceGroupManager_stateful </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/746576fa-b13f-447f-a35d-89c7c026e5af/recording/TestAccFirebaserulesRelease_BasicRelease.log)] 

Tests failed during RECORDING mode:
 `TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/746576fa-b13f-447f-a35d-89c7c026e5af/build-log/recording_build/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/746576fa-b13f-447f-a35d-89c7c026e5af/recording/TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample.log)]
`TestAccRegionInstanceGroupManager_stateful`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/746576fa-b13f-447f-a35d-89c7c026e5af/build-log/recording_build/TestAccRegionInstanceGroupManager_stateful_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/746576fa-b13f-447f-a35d-89c7c026e5af/recording/TestAccRegionInstanceGroupManager_stateful.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/746576fa-b13f-447f-a35d-89c7c026e5af/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7025/artifacts/746576fa-b13f-447f-a35d-89c7c026e5af/recording) for each test

bschaatsbergen commented:
Fixed the previous mentioned issue @slevenick, I don't have an org so it's a bit trial and error as I can't see the Cloud Build output of the VCR tests

slevenick commented:
Error message here:
```
=== CONT  TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample
    provider_test.go:315: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_identity_platform_config.default will be updated in-place
          ~ resource "google_identity_platform_config" "default" {
                id                         = "projects/project/config"
                name                       = "projects/####/config"
                # (2 unchanged attributes hidden)
        
              - monitoring {
                }
        
                # (1 unchanged block hidden)
            }
        
          # google_identity_platform_tenant.tenant will be updated in-place
          ~ resource "google_identity_platform_tenant" "tenant" {
                id                       = "projects/project/tenants/tenant-bla1a"
                name                     = "tenant-bla1a"
                # (5 unchanged attributes hidden)
        
              + monitoring {
                  + request_logging {
                      + enabled = true
                    }
                }
            }
        
```

bschaatsbergen commented:
> Error message here:
> 
> ```
> === CONT  TestAccIdentityPlatformTenant_identityPlatformTenantMonitoringExample
>     provider_test.go:315: Step 1/2 error: After applying this test step, the plan was not empty.
>         stdout:
>         
>         
>         Terraform used the selected providers to generate the following execution
>         plan. Resource actions are indicated with the following symbols:
>           ~ update in-place
>         
>         Terraform will perform the following actions:
>         
>           # google_identity_platform_config.default will be updated in-place
>           ~ resource "google_identity_platform_config" "default" {
>                 id                         = "projects/project/config"
>                 name                       = "projects/####/config"
>                 # (2 unchanged attributes hidden)
>         
>               - monitoring {
>                 }
>         
>                 # (1 unchanged block hidden)
>             }
>         
>           # google_identity_platform_tenant.tenant will be updated in-place
>           ~ resource "google_identity_platform_tenant" "tenant" {
>                 id                       = "projects/project/tenants/tenant-bla1a"
>                 name                     = "tenant-bla1a"
>                 # (5 unchanged attributes hidden)
>         
>               + monitoring {
>                   + request_logging {
>                       + enabled = true
>                     }
>                 }
>             }
>         
> ```

Hmm still the same, got any idea what the cause might be?

slevenick commented:
> Hmm still the same, got any idea what the cause might be?

Still looks like the monitoring block isn't coming back from the API, so we need to handle that somehow. Might need ignore_read on the field, though then we just have to trust that we set the field correctly

slevenick commented:
Were there changes since the last review pass? I don't see any past a merge. We'll need a way to ignore read on the field that doesn't get returned by the API

bschaatsbergen commented:
Hey @slevenick, nope I haven't changed anything actually- I got stuck after the last review, I'm a bit astonished that 1 test passes and 1 doesn't. If it really isn't returned from the API, it's quite inconsistent. We could optionally ignore the field when reading indeed. What do you suggest we do?

slevenick commented:
> Hey @slevenick, nope I haven't changed anything actually- I got stuck after the last review, I'm a bit astonished that 1 test passes and 1 doesn't. If it really isn't returned from the API, it's quite inconsistent. We could indeed ignore read the field. What do you suggest we do?

Yeah that would be really inconsistent, but totally possible! I can try to bring it up to the internal team and see whats going on here.

bschaatsbergen commented:
I guess we want to close this and pick this in the future up again @slevenick ? Feel free to close it if so.
